lib/compress/zstd_compress_literals.h

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2016-present, Yann Collet, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under both the BSD-style license (found in the
+ * LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ * in the COPYING file in the root directory of this source tree).
+ * You may select, at your option, one of the above-listed licenses.
+ */
+
+#ifndef ZSTD_COMPRESS_LITERALS_H
+#define ZSTD_COMPRESS_LITERALS_H
```

Comment:
`zstd_compress_literals.c` only contains "special cases", RLE and noCompress,
but the "normal" huffman compression is elsewhere.
Why ?

Reply:
It was because ZSTD_compressLiterals depends on ZSTD_minGain which isn't literals specific. In order to prevent circular dependency, I moved ZSTD_minGain to zstd_compress_internal.h at which point I ran into static function not being used warning. 

I think can do the similar approach with MEM_STATIC to suppress the warning. Let me do that!